contents/tests/test_common.py

```diff
@@ -0,0 +1,168 @@
+        self.assertEqual('default', data['namespace'])
+        self.assertIsNone(data['container_name'])
+        [actual_name, actual_namespace, actual_container] = common.get_core_node_parameter_list()
+        self.assertEqual(data['name'], actual_name)
+        self.assertEqual(data['namespace'], actual_namespace)
+        self.assertEqual(data['container_name'], actual_container)
+
+        name = 'Node'
+        namespace = 'Space'
+        container = 'Container'
+        os.environ.setdefault('RD_NODE_DEFAULT_NAME', name)
+        os.environ.setdefault('RD_NODE_DEFAULT_NAMESPACE', namespace)
+        os.environ.setdefault('RD_NODE_DEFAULT_CONTAINER_NAME', container)
+        self.assert_dictionary_matches(name, namespace, container)
+
+        name = 'Config Node'
+        namespace = 'Config Space'
+        config_container = 'Config Container'
+        os.environ.setdefault('RD_CONFIG_NAME', name)
+        os.environ.setdefault('RD_CONFIG_NAMESPACE', namespace)
+        os.environ.setdefault('RD_CONFIG_CONTAINER_NAME', config_container)
+        self.assert_dictionary_matches(name, namespace, container)
+
+    def assert_dictionary_matches(self, name, namespace, container):
+        """Asserts that a """
+        data = common.get_code_node_parameter_dictionary()
+        self.assertEqual(name, data['name'])
+        self.assertEqual(namespace, data['namespace'])
+        self.assertEqual(container, data['container_name'])
+        [actual_name, actual_namespace, actual_container] = common.get_core_node_parameter_list()
+        self.assertEqual(name, actual_name)
+        self.assertEqual(namespace, actual_namespace)
+        self.assertEqual(container, actual_container)
+
+    def test_create_volume(self):
+        data = {}
+        self.assertIsNone(common.create_volume(data))
+
+        data = {'name': 'volumeName'}
+        self.assertEqual(data['name'], common.create_volume(data).name)
+
+        claim = 'CLAIM'
+        path_name = 'PATH'
+        path_type = 'Directory'
+        data = {
+            'name': 'volumeName',
+            'persistentVolumeClaim': {'claimName': claim},
+            'hostPath': {'path': path_name, 'type': path_type},
+        }
+        volume = common.create_volume(data)
+        pvc = volume.persistent_volume_claim
+        path = volume.host_path
+        self.assertEqual(claim, pvc.claim_name)
+        self.assertEqual(path_name, path.path)
+
+        data['hostPath'] = {'path': path_name}
+        volume = common.create_volume(data)
+        self.assertEqual(claim, volume.persistent_volume_claim.claim_name)
+        # This is strange -- the code is forcing a constraint on path type
+        # that kubernetes itself does not have. I think this assertion should
+        # pass but it does not. I think the indent on line 254 of common.py
+        # is wrong. So I'm going to stop here and ask the maintainer via
+        # a pull request.
+        # self.assertEqual(path_name, path.path)
```

Comment:
This is the assertion I was asking about. The line number has changed

Look at line 200 of common.py on master in https://github.com/rundeck-plugins/kubernetes/commit/950c40f4eb482848a47305a72be38b8ce2436a1f:

```
        # hostpath
        if "hostPath" in volume_data and "path" in volume_data["hostPath"]: 
           host_path = client.V1HostPathVolumeSource(
                path=volume_data["hostPath"]["path"]
            )
            if "hostPath" in volume_data and "type" in volume_data["hostPath"]:
                host_path.type = volume_data["hostPath"]["type"]
                volume.host_path = host_path
```

Line 200 is the last line.

Because it is indented to the same level as the line before, volume.host_path will only get set if type is defined in volume_data["hostPath"].

This constraint is not required by the k8s API and doesn't make sense in the logic it looks like you are trying to establish.

I can't be sure, but it seems to me you were looking for something more like this:

```
        # hostpath
        if "hostPath" in volume_data and "path" in volume_data["hostPath"]: 
           host_path = client.V1HostPathVolumeSource(
                path=volume_data["hostPath"]["path"]
            )
            if "type" in volume_data["hostPath"]:
                host_path.type = volume_data["hostPath"]["type"]
            volume.host_path = host_path
```

I can submit a separate PR for this change -- it won't have testing since that is what I'm adding here...but maybe it will make it clearer.

Reply:
See #107 for a PR reflecting this issue